lib/components/artwork_grids/infinite_scroll_grid.js

```diff
@@ -88,64 +61,45 @@ class InfiniteScrollArtworksGrid extends React.Component {
 
   constructor(props) {
     super(props)
+
     this.state = {
       sectionDimension: 0,
-      artworks: this.props.artworks,
-      page: this.props.artworks.length ? 1 : 0,
       completed: false,
       fetchingNextPage: false,
     }
 
     this._sentEndForContentLength = null
-  }
-
-  // Initial setup
-  componentWillMount(){
-    if (this.state.artworks.length === 0) {
-      this.fetchNextPage()
+    if (this.props.filter) {
```

Comment:
This is the only semi-annoying thing here, we set variables on construction which as far as I can tell does not trigger any additional queries, it's just....not as nice as it could be.